CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.5)
+cmake_minimum_required(VERSION 3.4.3)
```

Comment:
The main point of this cmake file and the stage1 source code is to bootstrap from C++. So supporting a wide range of C++ compiler versions and cmake versions is part what this accomplishes. Requiring a significantly newer cmake is a big problem.